version: "3.9"

services:
  postgres:
    image: postgres:16
    container_name: chatbot_postgres
    restart: always
    environment:
      POSTGRES_USER: markus
      POSTGRES_PASSWORD: 1234
      POSTGRES_DB: Crecenia_chatbot
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U markus"]
      interval: 5s
      timeout: 5s
      retries: 10
    networks:
      - chatbot_net

  pgadmin:
    image: dpage/pgadmin4:latest
    container_name: chatbot_pgadmin
    restart: always
    depends_on:
      postgres:
        condition: service_healthy  
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@crecenia.ai
      PGADMIN_DEFAULT_PASSWORD: 1234
      PGADMIN_CONFIG_SERVER_MODE: "False"
    ports:
      - "8081:80"
    volumes:
      - pgadmin_data:/var/lib/pgadmin
    networks:
      - chatbot_net

networks:
  chatbot_net:
    driver: bridge

volumes:
  postgres_data:
  pgadmin_data:
